src/main/java/com/ibm/eventstreams/kafkaconnect/plugins/xml/engines/StructToXmlBytes.java

```diff
@@ -233,17 +235,19 @@ private void processStruct(Document doc, Element parentElement, Struct source) {
 
                 case BYTES:
                     final Element bytesElement = doc.createElement(field.name());
-                    if (source.getBytes(field.name()) != null) {
-                        final byte[] bytes = source.getBytes(field.name());
-
+                    final byte[] bytes =
+                            Decimal.LOGICAL_NAME.equals(fieldSchema.name()) ?
```

Comment:
Do we need to have the `if` check for the schema name twice? (here and on line 246) 

Can this be structured so we only need to check this once?

Reply:
Updated